src/fsharp/AsyncLazy.fsi

```diff
@@ -0,0 +1,22 @@
+namespace Internal.Utilities.Library
+
+[<RequireQualifiedAccess>]
+module internal AsyncLazy =
+
+    /// Allows to specify the language for error messages
+    val SetPreferredUILang : preferredUiLang: string option -> unit
+
+/// Lazily evaluate the computation asynchronously, then strongly cache the result.
+/// Once the result has been cached, the computation function will also be removed, or 'null'ed out, 
+///     as to prevent any references captured by the computation from being strongly held.
+/// The computation will only be canceled if there are no outstanding requests awaiting a response.
```

Comment:
Please provide an alternative, simpler version where this assumption changes to this?

> "If the computation is canceled, and there are further outstanding requests awaiting a response, then the next caller re-runs the computation with its cancellation token. The first caller runs the computation synchronously on its own callstack with an unchanged cancellation token."

This will surely give much better debugging, and as far as I can see gives us the performance we had before?